### PR TITLE
Use 'PlainText' when writing to a host that doesn't support VT

### LINKED ISSUE #15978: `$PSStyle.Formatting.TableHeader` writes out the ANSI codes in a legacy console that doesn't support virtual terminal
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Start PowerShell 7.2.0-preview.9 with a legacy console host (e.g. use 2012R2), and then run `Get-Module`.

Expected: The table view should not have any ANSI code written out.

Actual: ANSI codes are written out in the table view

issue-preview 9

### Expected behavior

```console
The table view should not have any ANSI code written out.
ANSI code should be ignored when the host doesn't support virtual terminal.
```

### Actual behavior

```console
ANSI codes are written out in the table view
```

### Error details

_No response_

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.0-preview.9
PSEdition                      Core
GitCommitId                    7.2.0-preview.9
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #15978

Update the `ShouldOutputPlainText` to take care of the situation when we are writing to a host that doesn't support VT.

I manually verified that it suppresses the VT sequences on legacy console host:

image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs [hunk 1]
```diff
@@ -248,6 +248,11 @@ private static bool ShouldOutputPlainText(bool isHost, bool? supportsVirtualTerm
                         break;
                 }
             }
+            else if (isHost)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:251]: Can't we just change line 237 to default to PlainText?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:251]: That's even better.

**Review by SteveL-MSFT — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs [hunk 2]
```diff
@@ -234,7 +234,7 @@ public virtual void WriteLine(ConsoleColor foregroundColor, ConsoleColor backgro
 
         private static bool ShouldOutputPlainText(bool isHost, bool? supportsVirtualTerminal)
         {
-            var outputRendering = OutputRendering.Ansi;
+            var outputRendering = OutputRendering.PlainText;
 
             if (supportsVirtualTerminal != false)
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: Nit:
```suggestion
            if (supportsVirtualTerminal)
```

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: It's of the `bool?` type, so `!= false` is required.

Comment (iSazonov) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: It is `public virtual bool SupportsVirtualTerminal { get { return false; } }`

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: The parameter is `bool? supportsVirtualTerminal`, you can check out how this method is used.

Comment (iSazonov) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: > you can check out how this method is used.

I just do not see why it is made nullable - bool is enough in all methods.

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: Well, when it's not used by a host (`isHost == false`), I guess `supportsVirtualTerminal` doesn't make sense?
But feel free to open an issue for discussion if you want.

Comment (iSazonov) [reply src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs:239]: If follow the logic we should remove isHost parameter at all, shouldn't we? :-)

**Pull request merged**